WordPress/wporg-mu-plugins
# Add a Notice Block

The new redesign uses a different style of callout box from the hardcoded `.callout` or shortcode previously used. This PR adds a "notice" block to render the 4 different notice styles.

- Tip, green background with ℹ️  icon
- Info, blue background with ❓ icon
- Alert, yellow background with ⚠️  icon
- Warning, red background with ❌  icon

I forked this block from [one I'd written as a demo](https://github.com/ryelle/notice-block/) for [a WordCamp talk](https://ryelle.codes/wcmontclair-2022/).

Having a Notice block will let us keep the 4 different notice styles consistent, rather than hardcoding in background colors on group blocks, etc. This also restricts the content that can be in a notice, to act as guardrails to keep notices to the point.

| Frontend | Editor |
|------|-------|
| ![notices](https://user-images.githubusercontent.com/541093/213037223-64181e3c-bd30-4592-8db6-c15dafaab8da.png) | <img width="735" alt="Screenshot 2023-01-17 at 7 03 57 PM" src="https://user-images.githubusercontent.com/541093/213038873-e7461ddb-9edb-4d7e-9b8c-e1f730bebb71.png"><img width="454" alt="Screenshot 2023-01-17 at 6 52 29 PM" src="https://user-images.githubusercontent.com/541093/213037586-f71c8dbd-d11f-44c7-b5a5-6f42c0e67608.png"> |

We can use it in code/templates like this:

```html

<div class="wp-block-wporg-notice is-alert-notice">
<div class="wp-block-wporg-notice__icon"></div>
<div class="wp-block-wporg-notice__content"><p>[Notice content here.]</p></div>
</div>

```

To change the type, you need to change both the attribute and the class name. To change the text, just enter content in the paragraph. There can be multiple `p`s (though it's not likely).

(It's not a dynamic block because it was built editor-first, but if we hate that it can be switched to dynamic)

**To test:**

- Try adding the block to some post content, it should not break
- Changing the type should work
- Try adding the block to a template
- It should continue to work

The block does support padding, margin, and different font sizes, just in case we need those.
